### PR TITLE
Add `can_be_replaced_by` to gone items.

### PR DESCRIPTION
**Pull request by elliotcm**
Part of https://trello.com/c/pULByuHs/393-fix-invalid-content-store-data

Related to https://github.com/alphagov/whitehall/pull/2373

### CODE REVIEW

File: dist/formats/gone/publisher/schema.json [hunk 1]
```diff
@@ -43,12 +43,33 @@
           }
         }
       }
+    },
+    "links": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "can_be_replaced_by": {
+          "description": "The optional content_item that this gone item has replaced.",
```
Comment (jamiecobbett) [inline dist/formats/gone/publisher/schema.json]: Would this read better as: `Optional. The content_item that this gone item has replaced.`?

Comment (elliotcm) [inline dist/formats/gone/publisher/schema.json]: Sure.  Updated.

Comment (danielroseman) [inline dist/formats/gone/publisher/schema.json]: Looks like this needs updating in two other places.

Comment (elliotcm) [inline dist/formats/gone/publisher/schema.json]: @danielroseman Done, and rebased.

**Pull request merged**